--- conduit/modules/FacebookModule/pyfacebook/__init__.py	2007-12-30 22:37:14.000000000 +1300
+++ conduit/modules/FacebookModule/pyfacebook/__init__.py	2008-01-18 09:44:47.000000000 +1300
@@ -42,13 +42,6 @@
         from django.utils import simplejson
         RESPONSE_FORMAT = 'JSON'
     except ImportError:
-        print 'NOTE: PyFacebook can use simplejson if it is installed, which'
-        print 'is much faster than XML and also uses less bandwith. Go to'
-        print 'http://undefined.org/python/#simplejson to download it, or do'
-        print 'apt-get install python-simplejson on a Debian-like system.'
-        print ''
-        print 'Falling back to XML...'
-
         from xml.dom import minidom
         RESPONSE_FORMAT = 'XML'
 
